### DISCUSSION TITLE
Reload user roles

### DISCUSSION THREAD

**Discussion by dgiorgio** [gravitational/teleport#36808] [Q&A] [answered]
Is it possible to reload user roles without logging back into the teleport?
When a role is removed, the user is unable to use teleportation without logging in again.

**Comment by webvictim** [comment, marked as answer]
If the content of a role the user has is changed on the server, the cache of these roles on proxies/agents should be updated in real time so the permission changes take effect immediately.

If the user is assigned new roles or has existing roles removed, then a full logout/login cycle is required to refresh the "set" of roles that the user has.

There is a feature request here to be able to "refresh" the session without logging out: https://github.com/gravitational/teleport/issues/28368
